tokens/src/commands.rs

```
@@ -296,20 +311,26 @@ fn distribute_allocations(
         messages.push(message);
         stake_extras.push((new_stake_account_keypair, lockup_date));
     }
+    Ok(())
+}
 
-    let num_signatures = messages
-        .iter()
-        .map(|message| message.header.num_required_signatures as usize)
-        .sum();
-    if args.spl_token_args.is_some() {
-        check_spl_token_balances(num_signatures, allocations, client, args, created_accounts)?;
-    } else {
-        check_payer_balances(num_signatures, allocations, client, args)?;
-    }
-
+fn send_messages(
+    client: &RpcClient,
+    db: &mut PickleDb,
+    allocations: &[Allocation],
+    args: &DistributeTokensArgs,
+    exit: Arc<AtomicBool>,
+    messages: Vec<Message>,
+    stake_extras: StakeExtras,
+) -> Result<(), Error> {
     for ((allocation, message), (new_stake_account_keypair, lockup_date)) in
         allocations.iter().zip(messages).zip(stake_extras)
     {
+        if exit.load(Ordering::SeqCst) {
+            db.dump()?;
+            #[cfg(not(test))]
```

Why not return an error instead?

---

Oh! That does seem nice.

---

Updated